libs/libcommon/src/libcommon/simple_cache.py

```diff
@@ -142,7 +156,10 @@ def upsert_response(
     progress: Optional[float] = None,
     updated_at: Optional[datetime] = None,
 ) -> None:
-    CachedResponseDocument.objects(kind=kind, dataset=dataset, config=config, split=split).upsert_one(
+    records = CachedResponseDocument.objects(kind=kind, dataset=dataset, config=config, split=split)
+    decrease_metrics(records=records)
```

Comment:
Hmmm, good point. Is it possible to have more than one result in `records`? If it's only 0 or 1, the logic is OK, if it can be >1, it would not work anymore, since upsert_one only updates the first result.

Reply:
No, it should not be more than one since we have the constrain:
`    kind = StringField(required=True, unique_with=["dataset", "config", "split"])`

I will change to decrease only by one record.

Reply:
Maybe we should add a test on the length of records, and raise an error if it's more than 1. Just to be sure that if we drop the constraint one day, we'll see that it breaks the logic here.